src/gluonts/torch/model/deepar/deepar_network.py

```diff
@@ -0,0 +1,430 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+import pytorch_lightning as pl
+
+from gluonts.core.component import validated
+from gluonts.itertools import prod
+from gluonts.time_feature import get_lags_for_frequency
+from gluonts.torch.modules.distribution_output import (
+    DistributionOutput,
+    StudentTOutput,
+)
+from gluonts.torch.modules.scaler import MeanScaler, NOPScaler
+from gluonts.torch.modules.feature import FeatureEmbedder
+from gluonts.torch.util import weighted_average
+
+
+class DeepARNetwork(nn.Module):
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        num_feat_dynamic_real: int,
+        num_feat_static_real: int,
+        num_feat_static_cat: int,
+        cardinality: List[int],
+        embedding_dimension: Optional[List[int]] = None,
+        num_layers: int = 2,
+        num_cells: int = 40,
+        cell_type: str = "LSTM",
+        context_length: Optional[int] = None,
+        distr_output: DistributionOutput = StudentTOutput(),
+        dropout_rate: float = 0.1,
+        lags_seq: Optional[List[int]] = None,
+        scaling: bool = True,
+        num_parallel_samples: int = 100,
+    ) -> None:
+        super().__init__()
+        self.num_layers = num_layers
+        self.num_cells = num_cells
+        self.cell_type = cell_type
+        self.context_length = context_length or prediction_length
+        self.prediction_length = prediction_length
+        self.dropout_rate = dropout_rate
+        self.num_feat_dynamic_real = num_feat_dynamic_real
+        self.num_feat_static_cat = num_feat_static_cat
+        self.num_feat_static_real = num_feat_static_real
+        self.cardinality = cardinality
+        self.embedding_dimension = (
+            embedding_dimension
+            if embedding_dimension is not None or self.cardinality is None
+            else [min(50, (cat + 1) // 2) for cat in self.cardinality]
```

Comment:
I think we can maybe keep this in the network only. What I’m trying to achieve here is to have as much defaults as possible in the network directly, so that one can use it as a standalone object (with custom data loading, training procedure, whatever)

Reply:
Good point.